src/services/userService.ts

```diff
@@ -0,0 +1,152 @@
+'use server';
+import bcrypt from 'bcrypt';
+
+import connectDB from '../lib/dbConnect';
+import { stringToObjectId } from '../lib/utils';
+
+import TypeModel from '@/models/Type';
+import UserModel, { IUser } from '@/models/User';
+
+export async function getUsers() {
+    try {
+        await connectDB();
+
+        const users = await UserModel.find<IUser>({}, 'name');
+
+        return {
+            users,
+        };
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function createUser(name: string, password: string) {
+    try {
+        await connectDB();
+
+        const hashedPassword = await bcrypt.hash(password, 10);
+
+        const user: IUser = await UserModel.create({ name, password: hashedPassword });
+
+        await TypeModel.create({ name: 'Default', user, isDefault: true });
+
+        return {
+            user: {
+                name: user.name,
+                _id: user._id,
+            },
+        };
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function getUser(userID = '123456789012') {
+    try {
+        await connectDB();
+        // TODO: Get USER_ID from session
+        const userId = stringToObjectId(userID);
+
+        if (!userId) {
+            return { error: 'User not found' };
+        }
+
+        const user = await UserModel.findById<IUser>(userId, 'name');
+        if (user) {
+            return {
+                user,
+            };
+        } else {
+            return { error: 'User not found' };
+        }
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function authenticateUser(name: string, password: string) {
+    try {
+        await connectDB();
+
+        const user = await UserModel.findOne<IUser>({ name });
+        if (!user) {
+            return { error: 'User not found' };
+        }
+        return { result: await bcrypt.compare(password, user.password) };
+    } catch (error) {
+        return { error: 'Deletion failed' };
```

Comment:
Error message is not correct.

Reply:
Thank you for pointing this out. Updated: https://github.com/tsoibet/kaado/pull/8/commits/17fe8e180ecfad82b441055e8f5a3a96c360d415